File: src/main/kotlin/leetCode/medium/PalindromicSubstrings.kt
package leetCode.medium

import org.junit.jupiter.api.Assertions.assertEquals

class PalindromicSubstrings {
    fun countSubstrings(s: String): Int {
        TODO()
    }
}

fun main(args: Array<String>) {
    val solution = PalindromicSubstrings()

    val example1 = solution.countSubstrings("abc")
    assertEquals(3, example1)

    val example2 = solution.countSubstrings("aaa")
    assertEquals(6, example2)
}

File: go/leetCode/medium/PalindromicSubstrings.go
package medium

func countSubstrings(s string) int {
	count := 0

	for i := 0; i < len(s); i++ {
		var left = i
		var right = i

		for left >= 0 && right < len(s) && s[left] == s[right] {
			left--
			right++
			count++
		}

		left = i
		right = i + 1
		for left >= 0 && right < len(s) && s[left] == s[right] {
			left--
			right++
			count++
		}
	}

	return count
}
